ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncStep.java

```diff
@@ -69,13 +69,20 @@ protected CompletableFuture<List<BlockHeader>> requestHeaders(final Hash hash) {
 
     final int batchSize = context.getBatchSize();
     LOG.debug("Requesting headers for hash {}, with batch size {}", hash, batchSize);
+    final Optional<BlockHeader> maybeFinalizedHeader =
+        context
+            .getProtocolContext()
+            .getBlockchain()
+            .getFinalized()
+            .flatMap(context.getProtocolContext().getBlockchain()::getBlockHeader);
 
     final RetryingGetHeadersEndingAtFromPeerByHashTask
         retryingGetHeadersEndingAtFromPeerByHashTask =
             RetryingGetHeadersEndingAtFromPeerByHashTask.endingAtHash(
                 context.getProtocolSchedule(),
                 context.getEthContext(),
                 hash,
+                maybeFinalizedHeader.map(BlockHeader::getNumber).orElse(0L),
```

Comment:
No need to revert this part, since backward sync is always on PoS, so you can statically set 0 here